### ISSUE TITLE
visual refresh

### ISSUE THREAD

**Issue by nilnilnil** [duckduckgo/ios-search-and-stories#8]

**Comment by ghost** [comment]
I'm not sure whether or not #12 should be included in this issue.

**Comment by nilnilnil** [comment]
Undermenu

![ddg ios 7 - menu](https://f.cloud.github.com/assets/347240/2322415/290a1712-a3b1-11e3-9c82-0ae65d6a77c8.jpg)

**Comment by nilnilnil** [comment]
PTR

![ddg ios 7 - pull to refresh](https://f.cloud.github.com/assets/347240/2322421/38884a2e-a3b1-11e3-9588-bd2738ec3999.jpg)

**Comment by nilnilnil** [comment]
Saved

![ddg ios 7 - saved option 2](https://f.cloud.github.com/assets/347240/2322433/4d193ef8-a3b1-11e3-969b-042e8d977b95.jpg)

**Comment by nilnilnil** [comment]
Home

![ddg ios 7 - stories 1a](https://f.cloud.github.com/assets/347240/2322439/56d7adda-a3b1-11e3-8592-f389013b1ef0.jpg)

**Comment by nilnilnil** [comment]
Story view

![ddg ios 7 - story detail](https://f.cloud.github.com/assets/347240/2322445/66ad4bca-a3b1-11e3-9651-85190699aaa7.jpg)

**Comment by quanganhdo** [comment]
@nilnilnil I find the blurry undermenu to be a bit too much. I prefer something darker like this Vesper screenshot:

![screen568x568](https://f.cloud.github.com/assets/18567/2326572/b7e13e42-a3e4-11e3-9d00-0a14d024d99c.jpg)

The rest looks great.

**Comment by ghost** [comment]
@nilnilnil 

These look great! 

I'm especially loving the new search/navigation bar. Sean mentioned you might want to use native UI for handling search results?

For the under menu, what's being blurred? Is it a static image?

**Comment by quanganhdo** [comment]
@micpringle it looks like the Home page is being blurred.

**Comment by nilnilnil** [comment]
Is more needed here to get started?

**Comment by sreilly** [comment]
Hi Caine, we're ready to go on it.  Just to confirm, are Mic and I to work on this or were you talking to Anh?

**Comment by nilnilnil** [comment]
I'm up for whatever!  I am hoping it evolves into an everyone project.

**Comment by quanganhdo** [comment]
I'd love to join as well.

**Comment by ghost** [comment]
I recommend then, for the immediate future at least, that we detail on here what each of us is working on, to coordinate our efforts.

@quanganhdo assuming you have no objections, I'll pick up the new under/over menu. It seems as good a place to begin as any. Also, I think it would be good if we could get some realtime communication going via Slack, Skype, Jabber, IM etc. Please let me know your preference and your timezone.

**Comment by quanganhdo** [comment]
Sounds good to me. Either Slack or Skype is fine. I'm in CDT for now.

**Comment by ghost** [comment]
@quanganhdo Cool. Ping Sean an email (sreilly@infinitekind.com) with your preferred email address for Slack and we'll get you setup.

**Comment by nilnilnil** [comment]
Can we do this out in the open on IRC or the DDG jabber server?  The others are exclusive.

**Comment by sreilly** [comment]
out in the open is certainly preferred.  I've rejoined the "ios" room on the jabber server.  Mic will rejoin it too.  @quanganhdo do you have a username on the DDG Jabber server?

**Comment by sreilly** [comment]
I'm also open to IRC.. is there a DDG IRC server we can all join?  That might be preferred since it seems even more open than jabber.

**Comment by nilnilnil** [comment]
We have a freenode channel, but could make an iOS specific one.

**Comment by sreilly** [comment]
I think we're all agreed that IRC is preferred. I just joined the #duckduckgo channel.  An iOS or even mobile app specific channel would be great.

**Comment by ghost** [comment]
@nilnilnil 

We're making some good progress on the new _OverMenu_ with live blur:

Here you can see it being dragged in, over the content:
![overmenu](https://f.cloud.github.com/assets/20023/2397570/148f83b2-a9ec-11e3-8430-07b5301a4031.png)

And he's a side-by-side comparison of your original screenshot alongside it running live on the device:
![comparison](https://f.cloud.github.com/assets/20023/2397572/203a9ef4-a9ec-11e3-8755-e3c2041147d5.png)
Pretty close :thumbsup: 

Please note that up until now we've been working solely on the interaction and the blur, which is why the status bar is still visible etc.

**Comment by nilnilnil** [comment]
Cool. I am looking forward to testing it!

**Comment by ghost** [comment]
@nilnilnil 

There's a new build available on TF that incorporates the new _OverMenu_ into the current app.

This was built from our experimental _visual-refresh_ branch, and as such the menu is currently empty, meaning only stories are available in this build. 

Having said that, it should still give you a nice idea of how the new menu looks and feels.

**Comment by nilnilnil** [comment]
Cool!  Looking forward to seeing some more stuff :).

**Comment by ghost** [comment]
@nilnilnil 

Are you able to provide me with the exact font name and sizes you've used in the _over menu_ screenshots above, along with the RGB values of the red, grey and black?

**Comment by nilnilnil** [comment]
@chrismorast can do that much better than I can.  Is it worth sticking with system default on font and estimates on color to prevent you being blocked?

**Comment by ghost** [comment]
@nilnilnil Yep. That's what I'll do for now :thumbsup:

**Comment by ghost** [comment]
@nilnilnil 

Are you also able to sort out getting hold of the new assets need for the new _over menu_? 

Specifically, I need the following:
- Cross (X)
- Plus (+)
- Magnifying glass
- Each of the menu icons (home, saved etc.)

Ideally they should be flat, without bezels or shadows, and in a single colour, usually grey. We apply the actual colour at runtime.

Once the _over menu_ is complete I'll also need the assets for the remaining changes. So it may be worth thinking about getting those done as well.

**Comment by chrismorast** [comment]
@micpringle I will start gathering the assets and info you need and get them over to you sometime today or tomorrow.

**Comment by ghost** [comment]
Superb! Thanks @chrismorast. 

You may also want to have a rethink about the colour used in the section header titles, as they can get swallowed up by the blur depending on the content underneath ...

![screenshot 2014 03 27 15 49 33](https://cloud.githubusercontent.com/assets/20023/2539408/c8723c12-b5c7-11e3-94f5-da36e11823b0.png)

**Comment by chrismorast** [comment]
@micpringle, good call. I think we can go a bit darker with them. Somewhere in between what they are now and what the menu item color is. 

Also, I'd like explore different options on where the menu comes in from. The initial thought was that it would slide up form the bottom but it's hard to gauge whether that will feel natural unless I'm able to test it. Would that be a possibility and an easy update if we decide to change it?

**Comment by ghost** [comment]
@chrismorast 

We can definitely experiment with different options, but sliding the menu up from the bottom would interfere with Control Centre, especially if you want to keep the _drag-to-open_ gesture. Sliding in from the left feels the most natural since that's where the _hamburger_ button is located, and you swipe left-to-right to go back through the navigation stack; so it makes sense that when there's no more history the same gesture opens the menu.

Are you able to download the builds from TestFlight? If so, give the latest a whirl. It currently slides in from the left. Ignore the opaque history items, I've not got around to those just yet :wink:

**Comment by chrismorast** [comment]
I've just set up test flight and will be able to test the next build.

**Comment by sreilly** [comment]
Hi @chrismorast you're now in the provisioning profile and should be able to install the build from testflight now.  Thanks!

**Comment by chrismorast** [comment]
Thanks @sreilly!
@micpringle here's a link to the assets you requested. https://www.dropbox.com/sh/k8rlndrzgxite1c/l85ybzwqpG
I imagine there will be more to send so let me know what I've missed.

**Comment by chrismorast** [comment]
Missed the PTR assets. Will post these as well as any additional elements to the same dropbox folder.

**Comment by ghost** [comment]
@chrismorast 

Thanks for these :thumbsup: 

The only remaining things I need for the _over menu_ is the RGB colour values and the fonts (_names, weights and sizes_).

**Comment by ghost** [comment]
@nilnilnil @chrismorast 

One thing we haven't discussed yet is how you'd like to handle deleting recent searches and stories. In the existing app it's done via a button which appears when swiping right-to-left, but a replacement isn't shown in any of the screenshots above.

**Comment by chrismorast** [comment]
@micpringle , I'll put together a comp for this. With regards to the color values and font specs, I'll get that over to you by EOD.

**Comment by chrismorast** [comment]
Here are two options I'm thinking of for the swipe options.
![swipe 2 share 6](https://cloud.githubusercontent.com/assets/7061920/2555040/9556b9fa-b6b9-11e3-854b-c34d91421559.jpg)
![swipe 2 share 3](https://cloud.githubusercontent.com/assets/7061920/2555041/955b39a8-b6b9-11e3-9559-3cabcab33241.jpg)

**Comment by chrismorast** [comment]
It would be nice to experiment with the orange background color from PTR in the stacked version as well.

**Comment by ghost** [comment]
@chrismorast 

Apologies if there's been some confusion, but I was referring to **swipe-to-delete**, in the context of the _over menu_ and deleting the history items.

![ddg](https://cloud.githubusercontent.com/assets/20023/2555117/8962609e-b6ba-11e3-992d-1a7dc6ab8564.png)

**Comment by chrismorast** [comment]
Got it... here's what I have for that. 
![swipe to delete1](https://cloud.githubusercontent.com/assets/7061920/2555386/4a24c242-b6be-11e3-8363-87fe9931b02d.jpg)

Realized that we can't use the orange for delete as that is the color for our official site badge. Using this instead for now

**Comment by ghost** [comment]
@chrismorast 

Ok, great. Could you provide the asset for the button? I doesn't need to contain the text, as I can add that at runtime (_and it'll be easier should you choose to localise the app at any point_).

**Comment by ghost** [comment]
@chrismorast 

Playing around with the latest build, I've been able to compile another list of new assets we'll need:
- Bang button
- Cancel button when searching
- Share icon when viewing a story
- Progress bar when searching/loading a web page
- Back arrow displayed in the navigation bar when searching

If I come across any more I'll add them.

**Comment by chrismorast** [comment]
Here's the delete button asset.
![delete](https://cloud.githubusercontent.com/assets/7061920/2570293/9365eac2-b8f7-11e3-9548-1c6056703d6a.png)
![delete 2x](https://cloud.githubusercontent.com/assets/7061920/2570294/936821d4-b8f7-11e3-88a7-5ce5da822c06.png)

**Comment by chrismorast** [comment]
can you post the original assets here for the pending items you posted about so I can update with matched specs?

**Comment by ghost** [comment]
@chrismorast 

As requested, an archive containing the existing assets can be found [here](https://dl.dropboxusercontent.com/u/25376082/Existing%20Assets.zip). Note, all these assets can also be found in the repo.

And some screenshots showing where they're used in the app:

![ddg_1](https://cloud.githubusercontent.com/assets/20023/2577183/2bf28e9c-b97c-11e3-99f7-b0197d6afd0f.png)

![ddg_2](https://cloud.githubusercontent.com/assets/20023/2577184/31097d6e-b97c-11e3-9008-f23811d425d9.png)

![ddg_3](https://cloud.githubusercontent.com/assets/20023/2577185/355dbc72-b97c-11e3-92de-7e9f7bbb2eda.png)

**Comment by ghost** [comment]
@chrismorast @nilnilnil 

I might also be worthwhile having a think about how you want to update the following screens, since they're not included in any of the above.

![ios simulator screen shot 1 apr 2014 10 14 25](https://cloud.githubusercontent.com/assets/20023/2577334/5635c3b6-b97e-11e3-9ccf-ebafd6d78c1f.png)

![ios simulator screen shot 1 apr 2014 10 14 29](https://cloud.githubusercontent.com/assets/20023/2577339/5aaa4aca-b97e-11e3-9951-ba50e6826466.png)

![ios simulator screen shot 1 apr 2014 10 14 44](https://cloud.githubusercontent.com/assets/20023/2577341/5eac89c6-b97e-11e3-967b-c2fdb8721cc7.png)

**Comment by ghost** [comment]
@chrismorast 

I've re-implemented delete for the menu history items, using the asset you provided. Please check out the latest build from Hockey.

Do you think there should be some visual feedback for the user when they tap the button? At the moment it feels a little too static.

**Comment by ghost** [comment]
@chrismorast 

Something else I need is the PTR arrow icon.

![ptr](https://cloud.githubusercontent.com/assets/20023/2578886/55cc4d1c-b999-11e3-8843-1783e3c7d9f5.png)

If you could provide in the same way as you have with the others; grayscale, transparent background etc. Cheers!

**Comment by nilnilnil** [comment]
@micpringle this is coming along great.  Nice work!

**Comment by ghost** [comment]
Cheers @nilnilnil - it's been fun to work on :thumbsup:

**Comment by chrismorast** [comment]
@micpringle - Putting these assets together now. Should have them over to you sometime tomorrow mid to late morning (PDT). One quick note, there should be a 1pt drop shadow on the title frames with no spread, color #000 at .2 (or .25) opacity.

**Comment by ghost** [comment]
@chrismorast 

> One quick note, there should be a 1pt drop shadow on the title frames with no spread, color #000 at .2 (or .25) opacity.

Aha! I missed that, cheers. It's just been added; please check out the latest build on Hockey.

**Comment by chrismorast** [comment]
@micpringle here's the ptr assets
![ptr](https://cloud.githubusercontent.com/assets/7061920/2594254/92adb0d4-ba8b-11e3-97c1-c6ee853ffde8.png)
![ptr 2x](https://cloud.githubusercontent.com/assets/7061920/2594255/92b04df8-ba8b-11e3-8ad5-77ce2a97b610.png)

**Comment by chrismorast** [comment]
@micpringle , here's the bang screenshot update
![bang](https://cloud.githubusercontent.com/assets/7061920/2595906/47b92cce-baa0-11e3-9668-60c65cb41705.jpg)

**Comment by chrismorast** [comment]
Here are the assets. The "+" icon should be the same as what we've used in the menu.

![bang-loupe_icon 2x](https://cloud.githubusercontent.com/assets/7061920/2596047/31a72ede-baa2-11e3-9666-94c20d7fb709.png)
![bang_1 2x](https://cloud.githubusercontent.com/assets/7061920/2596048/31adffb6-baa2-11e3-8f37-6540587fb689.png)
![clear_icon 2x](https://cloud.githubusercontent.com/assets/7061920/2596049/31b3e52a-baa2-11e3-93f7-729400ecf15d.png)
![bang-loupe_icon](https://cloud.githubusercontent.com/assets/7061920/2596050/31b8c496-baa2-11e3-967d-8ec7f99418de.png)
![bang_1](https://cloud.githubusercontent.com/assets/7061920/2596051/31bed35e-baa2-11e3-81e8-7e284c747989.png)
![clear_icon](https://cloud.githubusercontent.com/assets/7061920/2596052/31c40d4c-baa2-11e3-836f-feb25dd5742f.png)

**Comment by chrismorast** [comment]
Here's the load bar assets. May have to explore colors here. Let's start with #bfdfff.
![load_bar 2x](https://cloud.githubusercontent.com/assets/7061920/2596207/30d25338-baa4-11e3-8c1f-f6c55e8ee9e2.png)
![load_bar](https://cloud.githubusercontent.com/assets/7061920/2596208/30e3a35e-baa4-11e3-9fe7-c542d639fd21.png)

**Comment by chrismorast** [comment]
…and the other bang screen. I'd like to remove the borders from the icons if that's possible.
![bang1](https://cloud.githubusercontent.com/assets/7061920/2596241/b01c0e4a-baa4-11e3-9636-82b87db1bc10.jpg)

**Comment by chrismorast** [comment]
tool tip:
![bang1-tool-tip](https://cloud.githubusercontent.com/assets/7061920/2596320/e77e7a8e-baa5-11e3-98af-5eec7a640f63.jpg)

**Comment by chrismorast** [comment]
Here's the back button asset...
![back 2x](https://cloud.githubusercontent.com/assets/7061920/2597470/08d7ea62-bab5-11e3-91e9-052e58f46d4f.png)
![back](https://cloud.githubusercontent.com/assets/7061920/2597471/08dba008-bab5-11e3-9480-9ca556089eb8.png)

**Comment by chrismorast** [comment]
Here's the share button asset.
![share](https://cloud.githubusercontent.com/assets/7061920/2598020/02974bee-babe-11e3-97c7-49300788ecc5.png)
![share 2x](https://cloud.githubusercontent.com/assets/7061920/2598021/02b802ee-babe-11e3-9cdd-6747c318209c.png)

**Comment by chrismorast** [comment]
Here are the swipe to share assets. 
![safari](https://cloud.githubusercontent.com/assets/7061920/2598121/c4d6a884-babf-11e3-937f-7de0ed146233.png)
![safari 2x](https://cloud.githubusercontent.com/assets/7061920/2598122/c4d99512-babf-11e3-83d6-e9a7d9632c4d.png)
![share-large](https://cloud.githubusercontent.com/assets/7061920/2598123/c4dfb532-babf-11e3-9754-eb95e2296bf6.png)
![add-to-favorites 2x](https://cloud.githubusercontent.com/assets/7061920/2598124/c4e51496-babf-11e3-8509-fca1c165dd8c.png)
![share-large 2x](https://cloud.githubusercontent.com/assets/7061920/2598125/c4eae89e-babf-11e3-8e92-488230863a5c.png)
![add-to-favorites](https://cloud.githubusercontent.com/assets/7061920/2598126/c4f02976-babf-11e3-8c9c-0a7538a14fe7.png)

**Comment by ghost** [comment]
Cheers @chrismorast :thumbsup:

**Comment by ghost** [comment]
@chrismorast 

It might be worth providing new assets as zip files, rather than pasting them onto this thread. I'm not sure whether or not GitHub does any sort of compression etc. on the images you upload, and if so then the ones I end up using won't be of the same quality as the ones you provide.

**Comment by ghost** [comment]
@chrismorast 

Should I be using a different _home_ icon for the search bar? I'm currently using the same one used in the _over menu_, but it doesn't look quite right now I've added the new share icon.

![screen shot 2014-04-03 at 13 29 07](https://cloud.githubusercontent.com/assets/20023/2603211/b90835e2-bb2b-11e3-9e4b-2c37d83b7893.png)

It feels as though it should be both thinner, and bigger.

**Edit:**

Here's a comparison of all the new search bar icons you've provided. The _Share_ icon is definitely the odd one out.

![icons](https://cloud.githubusercontent.com/assets/20023/2603435/0b34ab7c-bb2f-11e3-8f10-8b8ed81a52f7.png)

**Comment by ghost** [comment]
@chrismorast @nilnilnil 

Here's a breakdown on things I'm still waiting on:
- Feedback on the issue raised above re: the icons
- A comprehensive breakdown on the fonts and colours that have been used throughout
- Empty states for the following screens ...

![ios simulator screen shot 10 apr 2014 13 58 40](https://cloud.githubusercontent.com/assets/20023/2667661/b4b64080-c0b0-11e3-8452-5acd9393442d.png)

![ios simulator screen shot 10 apr 2014 13 58 46](https://cloud.githubusercontent.com/assets/20023/2667665/bd2586ea-c0b0-11e3-924e-cf653c77e17b.png)
- The new visuals for the following screens ...

![ios simulator screen shot 10 apr 2014 13 58 58](https://cloud.githubusercontent.com/assets/20023/2667675/d6683be8-c0b0-11e3-9d7a-45df41e79b1f.png)

![ios simulator screen shot 10 apr 2014 13 59 06](https://cloud.githubusercontent.com/assets/20023/2667677/dae75ac8-c0b0-11e3-93b1-fd2ff3302ba4.png)

I think once I have all this I should be able to wrap up the visual refresh.

-Mic

**Comment by chrismorast** [comment]
@micpringle you're right, share doesn't fit. Let's go back to the older style for now.
![share 2x](https://cloud.githubusercontent.com/assets/7061920/2699220/0bebf34e-c404-11e3-9b5e-1aae36bb8f07.png)
![share](https://cloud.githubusercontent.com/assets/7061920/2699221/0bee4856-c404-11e3-9848-9eeef1994fe3.png)

**Comment by chrismorast** [comment]
![ddg ios 7 - story detail copy](https://cloud.githubusercontent.com/assets/7061920/2699230/26daef8e-c404-11e3-8ea1-7ebe67b9243b.jpg)

**Comment by chrismorast** [comment]
Here's the new "favorites" (formerly "Saved") comp. Assets to follow.
![favorites](https://cloud.githubusercontent.com/assets/7061920/2699311/0872e4ba-c405-11e3-822a-c9573bb0224e.jpg)

**Comment by chrismorast** [comment]
![mini-share 2x](https://cloud.githubusercontent.com/assets/7061920/2699315/143eb4fe-c405-11e3-9f74-8af0d10a3d7e.png)
![no-favorites](https://cloud.githubusercontent.com/assets/7061920/2699316/14452df2-c405-11e3-829e-1a3006f8febd.png)
![no-favorites 2x](https://cloud.githubusercontent.com/assets/7061920/2699317/14474380-c405-11e3-8331-43f84fb97cfa.png)
![mini-share](https://cloud.githubusercontent.com/assets/7061920/2699318/14498640-c405-11e3-99d9-6a002298ff71.png)

**Comment by chrismorast** [comment]
Here's how the settings pages should be styled.
![settings](https://cloud.githubusercontent.com/assets/7061920/2700197/5caa85b4-c410-11e3-8dd2-24ae6bb2bd49.jpg)

**Comment by chrismorast** [comment]
@micpringle let me know if I've missed anything else.

**Comment by chrismorast** [comment]
Forgot this one - when stories are turned off.
![duck-mode](https://cloud.githubusercontent.com/assets/7061920/2700426/a859a36c-c412-11e3-83dd-7b0c40946ece.jpg)

**Comment by chrismorast** [comment]
Here's that logo asset.
![duck-mode](https://cloud.githubusercontent.com/assets/7061920/2700443/dd55f5de-c412-11e3-93bf-213977f66ce3.png)
![duck-mode 2x](https://cloud.githubusercontent.com/assets/7061920/2700444/dd5cc10c-c412-11e3-8500-3f5d3587e723.png)

**Comment by chrismorast** [comment]
@micpringle,  can you please change the menu so that it slides up from the bottom?

**Comment by nilnilnil** [comment]
@chrismorast if it's an easy change, I think it's a good thing; otherwise, it might be too much for this.  I know that @micpringle spent a few days on this slide out implementation.

**Comment by ghost** [comment]
@chrismorast 

We can quite easily change the menu so it slides up from the bottom, but I'd err on the side of caution. 

iOS' Control Center slides up from the bottom using a system-wide gesture which takes priority over all local gestures; there's a high chance DDG users will end up frustrated, wanting to bring up the menu but instead being presented with the Control Center; it's why it's not a common paradigm found in iOS outside of the OS itself.

Having the menu slide on from the right _feels_ much more natural; swiping left-to-right to go back through the web pages or views, and when there are no more it brings the menu into view. Having the menu slide up from the bottom would break that fluidity.

However, the choice is yours so please let me know if you'd like to proceed.

/cc @nilnilnil @sreilly

**Comment by chrismorast** [comment]
@micpringle let's have it slide up and turn off the swipe-to-open option on the menu. I'm hoping this would be easy to revert back if we decide we don't care for it after a hands on experience with it.

**Comment by ghost** [comment]
@chrismorast There's now a build available that has the menu sliding up from the bottom. If you decide you want to keep this implementation then I'd recommend designing a new menu icon since the _hamburger_ icon has become synonymous with the menu coming in from the left.

**Comment by ghost** [comment]
@chrismorast Another screen that appears to have been missed:

![recent](https://cloud.githubusercontent.com/assets/20023/2730488/5fc4ff8e-c61a-11e3-8d7f-604a0433d131.png)

This is the screen you get when setting your Home screen to Recent.

**Comment by ghost** [comment]
@chrismorast I also need to know how the highlighted state of the menu items should look:

![highlight](https://cloud.githubusercontent.com/assets/20023/2730512/edeedcee-c61a-11e3-9035-df0dcaa75b02.png)

**Comment by chrismorast** [comment]
Hi @micpringle , the app is coming along nicely!

I've made some new swipe option assets and would like you to match this attached screen.
Please note that the drop shadow that was under the title bar that overlays the story thumbnail is now an inner shadow on the orange background. Let me know if you have any questions. I'll be posting some menu updates as well.
![swipe-options](https://cloud.githubusercontent.com/assets/7061920/2784275/bb1b5a08-cb39-11e3-8004-c8b79f48402c.jpg)
![icon_add-to-favorites](https://cloud.githubusercontent.com/assets/7061920/2784282/e31a98ca-cb39-11e3-9688-8e709a435bf5.png)
![icon_add-to-favorites 2x](https://cloud.githubusercontent.com/assets/7061920/2784287/e355821e-cb39-11e3-8ab9-250a34d07fc6.png)
![icon_open-in-safari](https://cloud.githubusercontent.com/assets/7061920/2784284/e3436fb6-cb39-11e3-98c3-2d5bf7ae3040.png)
![icon_open-in-safari 2x](https://cloud.githubusercontent.com/assets/7061920/2784283/e3408ad0-cb39-11e3-8ab3-e8b884ded116.png)
![icon_share](https://cloud.githubusercontent.com/assets/7061920/2784285/e349af2a-cb39-11e3-9581-c4d21fcc6483.png)
![icon_share 2x](https://cloud.githubusercontent.com/assets/7061920/2784286/e34f6bf4-cb39-11e3-9f12-6b54412f9e3e.png)

**Comment by chrismorast** [comment]
Here's the updated menu comp

Please note that the X should be positioned at center with the menu icon. In addition, we'll have to align the menu items to that, adjust the margins on the right to match, and add some more padding between the main menu items and the X. Let me know if you have any questions. Thanks!
![new-menu](https://cloud.githubusercontent.com/assets/7061920/2784358/65713814-cb3b-11e3-84c2-0517b7ba0217.jpg)

**Comment by chrismorast** [comment]
One more thing @micpringle 
The text in the menu should left align with the text in the recent searches. In addition, recent search icons/favicons should be left aligned with the left of the menu item icons.

**Comment by sreilly** [comment]
Hi @chrismorast, would you mind if we returned the swipe-sideways gesture to show and hide the menu?  I find having that menu accessible via a swipe gesture _really_ handy.  Without it, it feels like a regression from the currently released version of the app.  It does mean making the menu come in from the side instead of the bottom, but I think that's expected since many apps work in that way with this kind of menu.

I think the gestures are important to make the app feel natural. This blog post from a well known iOS dev talks a bit about it: http://blog.jaredsinclair.com/post/83544603052/sloppy-swiping-how-to-make-an-app-feel-comfortable

I think of gestures kind-of like keyboard shortcuts.  They aren't always obvious (which is why there's the button) but power users will benefit greatly from them.

**Comment by ghost** [comment]
@chrismorast 

For the new swipe assets, we need a favourite icon with a minus sign for when someone has already saved the article.

This is what's used now:

![screen shot 2014-04-24 at 10 56 56](https://cloud.githubusercontent.com/assets/20023/2788207/1c579b08-cb97-11e3-8f49-b48a7dbf1620.png)

**Comment by ghost** [comment]
@chrismorast 

I've just pushed a build that contains all the changes to the menu that you've requested above.

**Comment by nilnilnil** [comment]
I have a side question @micpringle @sreilly what's entailed in replacing our custom 'layers' idea with the  native implementation?

**Comment by ghost** [comment]
@nilnilnil 

It would basically mean we'd have to re-architect the entire app. The current implementation is far from  ideal, and there's a lot of inter-dependencies between view controllers (_what I think you're referring to as layers_) so whilst we'd be able to recycle some of the legacy code, a lot of new code would also need to written.

It would be a significant investment on your part, and the benefits would mainly be non-visual. It would however set you up well for future revisions or additions, as well as put you in good stead should you wish to pursue an iPad version. It would also make it much easier to adopt the latest UI-related APIs in iOS 7 or the imminent iOS 8 release.

We'd obviously be happy to undertake the work, but we want to be clear about what's involved so you can make an informed decision.

**Comment by nilnilnil** [comment]
@micpringle well then bail on that until we fully redesign instead of just visual.  Thanks for the thoughtful reply.  Side note, why is there any legacy code?  Is some of it dead or unsupported?  Or you're just referring to legacy as old?

**Comment by ghost** [comment]
@nilnilnil No problem. 

Yeah `legacy == old`

**Comment by chrismorast** [comment]
Yeah @sreilly , I'm not digging the slide up menu either. I'd love to get away from the side slide-in menu but it may require more time for exploration than we have. So we can push that off until the full redesign. Let me discuss our options with the rest of the team and get back to you on how to proceed with the menu.

**Comment by chrismorast** [comment]
OK, so we all agree that we'd prefer the menu to slide in from the left (as @sreilly is recommending).

In addition we have a few more updates;
1. Change all orange throughout the app to match the background of the swipe options (I believe it's #cc6956 but love what ever it is you have in there). This includes the PTR background, the icons, the saved tabs (at the bottom) 
2. In DuckMode you can remove the "Stories" option from the menu
3. Can we try this attached icon - we've changed the color to be more analogous to the orange in listed above.

![duckduckgo_1024x1024](https://cloud.githubusercontent.com/assets/7061920/2795083/2cecab6c-cbf5-11e3-8c84-9a8a0fd2cc22.png)

**Comment by chrismorast** [comment]
@micpringle here's the remove-from-favorites icon
![icon_remove-from-favorites](https://cloud.githubusercontent.com/assets/7061920/2795124/c205e038-cbf5-11e3-80d7-6545d5939761.png)
![icon_remove-from-favorites 2x](https://cloud.githubusercontent.com/assets/7061920/2795125/c242270a-cbf5-11e3-9c9c-8201b0e35a5e.png)

**Comment by nilnilnil** [comment]
I don't agree with #2.  What's the rationale there?( phone )

**Comment by chrismorast** [comment]
it wasn't the intent... I'm okay with leaving it but I'll have to make a new icon for it

**Comment by ghost** [comment]
@chrismorast You may also want to think about giving these a visual facelift:

![toast](https://cloud.githubusercontent.com/assets/20023/2798573/71c31d78-cc51-11e3-9d70-e19b3bae1e43.png)

**Comment by ghost** [comment]
@chrismorast 

I've updated the app to use the colour you provided: #CC6956. However, the colour I actually had in the swipe menu was #DC5F4B. The difference is subtle. If you want to switch it back just let me know, but you'll also need to provide another app icon.

I use [Acorn](http://flyingmeat.com/acorn/) for determining and picking colours, with no colour profile attached. If you are using something like Photoshop, make sure you remove any colour profile otherwise there'll always be subtle differences between our colours. Either that, or provide a list of the exact colours you'd like to use for _all_ colours and we'll work from that, rather than picking them from your screenshots.

I still need a list of the fonts and font sizes as well.

**Comment by ghost** [comment]
@chrismorast 

Also, don't forget about the things mentioned in [this comment](https://github.com/duckduckgo/ios/issues/8#issuecomment-40700804) and [this comment](https://github.com/duckduckgo/ios/issues/8#issuecomment-40700569) :+1:

**Comment by chrismorast** [comment]
I'm having a hard time deciding which to go with in regards to color. So I've made a color right in between. Can you change all of the orange in the app to #d06355 - need to make sure we update the background of the Favorites Screen (when there are no favorites "You have no favorites")

**Comment by chrismorast** [comment]
we may want to update the icons to a different color though... just want to see that new color implemented first.

**Comment by chrismorast** [comment]
Here's the recents screen
![recents](https://cloud.githubusercontent.com/assets/7061920/2806600/8eb238a2-cccd-11e3-8edc-ecb4755871c6.jpg)

**Comment by chrismorast** [comment]
The stories would actually be stories though :)

**Comment by chrismorast** [comment]
Here's the menu click. It's color #000 @ .2 opacity with white text/icon
![menu-click](https://cloud.githubusercontent.com/assets/7061920/2806652/f9aa335c-ccce-11e3-8202-b18a91ad1708.jpg)

**Comment by chrismorast** [comment]
here's another option for the recents screen. @nilnilnil which do you prefer? They're in dropbox if you want to test on your phone (https://www.dropbox.com/home/DuckDuckGo%20Assets/Mobile/iOS/iOS%207/Dev%20Comps)

![recents1](https://cloud.githubusercontent.com/assets/7061920/2809386/976a5f58-cd6b-11e3-9fa8-d16bec29a785.jpg)

**Comment by ghost** [comment]
@chrismorast What colour should the background be in the Favourite Stories/Searches view?

![favorite stories](https://cloud.githubusercontent.com/assets/20023/2814614/e97ddad0-ceac-11e3-8396-dadd609e8f54.png)

_Edit_: This is actually the colour behind any view that displays stories, searches etc.

**Comment by ghost** [comment]
@chrismorast Is it also worth revisiting this screen, which is displayed when the app is first run?

![first run](https://cloud.githubusercontent.com/assets/20023/2814644/5c644a2a-cead-11e3-8a5c-bd58ba20e3c9.png)

**Comment by ghost** [comment]
@chrismorast I've just pushed a build that updates all the use of red to #D06355

**Comment by ghost** [comment]
@chrismorast I've just pushed another build that includes the changes you requested to the menu items' highlighted state :thumbsup:

**Comment by nilnilnil** [comment]
I think we should avoid super red screens.  It's just overwhelming.  I also agree we need to update the intro screen.  I think the BG color should be the light grey to match the button BG.

**Comment by chrismorast** [comment]
I agree @nilnilnil . @micpringle can you use the gray version of the recents screen? BTW, great work on everything so far. It's really coming together well.

**Comment by chrismorast** [comment]
@micpringle yes, we should update the 1st time run screen. I will post a comp and assets later today.

**Comment by chrismorast** [comment]
Here's the background color for the no-content areas. #e0e0e0 (color in screenshot is approximate - so it may be a little off)
![no-content_bg](https://cloud.githubusercontent.com/assets/7061920/2819369/a185a2b2-ceed-11e3-9cd1-6af2c7afc3d4.png)

**Comment by chrismorast** [comment]
Here's the updated splash screen (button color is #D06355 with a 2pt inner shadow on the bottom that's rgba 0,0,0,.15). Please use the same logo assets that were provided for DuckMode earlier in this thread. Happy to repost if you'd prefer that.
![splash](https://cloud.githubusercontent.com/assets/7061920/2825822/4e877cde-cf5b-11e3-9494-2fd98b7bc697.jpg)

**Comment by ghost** [comment]
> BTW, great work on everything so far. It's really coming together well.

Thanks @chrismorast :thumbsup:

**Comment by ghost** [comment]
@chrismorast I've updated the recents screen:

![recent](https://cloud.githubusercontent.com/assets/20023/2827100/674a3bae-cf7d-11e3-98cd-746bfeb3ae80.png)

However, I think it may need a little more work. As you can see, the highlighted state isn't as aesthetically pleasing here as it is in the menu (_different background_) and the font and font sizes feel way too small. Speaking of which, I _really_ could do with those details of the fonts used throughout the refresh.

**Comment by ghost** [comment]
@chrismorast For the Getting Started screen, I'll need you to provide the assets for the button.

Here's what we're currently using:

![btn_get-started](https://cloud.githubusercontent.com/assets/20023/2827149/6c33517c-cf7e-11e3-8c50-64d782feccaa.png)
![btn_get-started 2x](https://cloud.githubusercontent.com/assets/20023/2827151/6fc067a8-cf7e-11e3-9ed7-cb6ae69bf456.png)

Feel free to include the _Get Started_ text in the new assets, it'll make it easier to put together. Also, these button assets will need to be provided exactly how you'd like them to appear, **not** in grayscale like the icons etc.

**Comment by chrismorast** [comment]
Thanks @micpringle , I will get you an update on the recents screen, the button and text for the for the home screen as well as the font spec by tonight sometime.

**Comment by chrismorast** [comment]
Here are the button assets.
![btn_get-started](https://cloud.githubusercontent.com/assets/7061920/2833864/3523d4e2-cfd6-11e3-86db-341ce366e171.png)
![btn_get-started 2x](https://cloud.githubusercontent.com/assets/7061920/2833865/353eeb60-cfd6-11e3-8b96-04f62118230e.png)

**Comment by chrismorast** [comment]
@micpringle - here are the icon assets for Stories in the menu
![stories](https://cloud.githubusercontent.com/assets/7061920/2834406/7b9db036-cfdc-11e3-942c-e3f316893436.png)
![stories 2x](https://cloud.githubusercontent.com/assets/7061920/2834407/7bc31ca4-cfdc-11e3-88f1-b34c3e425147.png)

**Comment by chrismorast** [comment]
In regards to the on click state, can we try switching it to #D06355 with 100% opacity in both cases?

**Comment by chrismorast** [comment]
Need to do these screens as well.
![no history](https://cloud.githubusercontent.com/assets/7061920/2835621/5cc2f2ec-cfec-11e3-9aef-9c5a5caa26c3.jpg)
![history disabled](https://cloud.githubusercontent.com/assets/7061920/2835629/66f39c8a-cfec-11e3-9b33-2444a79ba0cb.jpg)
![icon_message](https://cloud.githubusercontent.com/assets/7061920/2835633/6f2442c4-cfec-11e3-946e-548c05686be5.png)
![icon_message 2x](https://cloud.githubusercontent.com/assets/7061920/2835634/6f272ac0-cfec-11e3-9685-4d7535bed63f.png)

**Comment by chrismorast** [comment]
As far as the fonts go, can we use Proxima Nova? I believe we purchased a license for it. If so, I'll get the sizes over to you. What's the best measurement for me to use, points?

**Comment by nilnilnil** [comment]
I think we only bought the web version for now.

**Comment by ghost** [comment]
@chrismorast We _could_ use Proxima Nova, but as @nilnilnil alludes to, you'll first need to purchase a license that allows you to embed the font in a mobile app. I believe [Font Spring](http://www.fontspring.com/fonts/mark-simonson-studio/proxima-nova) handles this for [Proxima Nova](http://www.fontspring.com/fonts/mark-simonson-studio/proxima-nova).

One thing to be aware of though is that the commercial license of the font may well conflict with the open-source license of the DDG app. We can certainly include the font in the shipping app once you've procured the appropriate licence, but you would need to make sure it's excluded from the DDG repo on GitHub, or anywhere else the codebase is publicly hosted.

**Comment by ghost** [comment]
@chrismorast 

> In regards to the on click state, can we try switching it to #D06355 with 100% opacity in both cases?

This actually works really well :thumbsup:

**Comment by chrismorast** [comment]
@micpringle Nice! Yeah, I was a little unsure of that myself :)

We have a few revisions/requests:
1. On visited stories, please change title color to #9e9e9e
2. Have all cookies cleared on app open
3. Add swipe away gesture to menu
4. Duckmode, do you have everything you need for that? Assets and screenshot is included above in this thread.

Can you tell me what size font you're using on the story titles?

**Comment by chrismorast** [comment]
One thing I'm noticing... the header drop shadow is really dark. i believe it's because it's doubled up with the PTR drop shadow.
The PTR drop shadow (as well as the story swipe menu drop shadow) should actually be an inner shadow. So it would overlay the orange color at the bottom, not just below that. Fixing that should clear it up. Let me know if that makes sense.

**Comment by ghost** [comment]
@chrismorast 

> Add swipe away gesture to menu

This is actually in the latest build. We've spent a lot of time recently refining it to make sure it feels _just right_, whilst not conflicting with things like swipe-to-delete on the history items etc. Give it a shot!

**Comment by ghost** [comment]
@chrismorast 

> Can you tell me what size font you're using on the story titles?

It's currently Helvetica Neue at 14pts.

**Comment by ghost** [comment]
@chrismorast 

> Have all cookies cleared on app open

It appears this has already been fixed and merged into _your_ codebase. See #15. ~~It's currently not in ours since we're working from our own fork. I will look to see if I can merge it in though.~~ This is now merged into our fork so you should see the functionality present in our builds.

**Comment by ghost** [comment]
@chrismorast 

> One thing I'm noticing... the header drop shadow is really dark. i believe it's because it's doubled up with the PTR drop shadow.

It appears as though our inner shadow is conflicting with the 0.5pt drop shadow automatically added to the navigation bar by the OS.

![shadow](https://cloud.githubusercontent.com/assets/20023/2851342/ec10804c-d116-11e3-8a86-c4fb7b047941.png)

Despite what the [docs state](https://developer.apple.com/library/ios/documentation/uikit/reference/uinavigationbar_class/Reference/UINavigationBar.html#//apple_ref/doc/uid/TP40006887-CH3-SW24), there seems to be no reliable way to remove the system drop shadow (_an iOS 7 bug perhaps?_). 

So as a workaround, I've changed the inner shadow to 0.5 pts, made it a little darker, and made it's display configurable. Then I simply don't display it for the first story cell. Give it a shot and let me know what you think.

![shadow2](https://cloud.githubusercontent.com/assets/20023/2851395/3c2a431e-d118-11e3-9d76-ef796afd3505.png)

**Comment by ghost** [comment]
@chrismorast 

The screens shown in [this comment](https://github.com/duckduckgo/ios/issues/8#issuecomment-41739531) have now been updated :thumbsup:

**Comment by ghost** [comment]
@chrismorast 

> Duckmode, do you have everything you need for that? Assets and screenshot is included above in this thread.

I _think_ I have everything I need. The implementation of Duckmode still needs some work, so if I come across anything I'll let you know.

**Comment by chrismorast** [comment]
@micpringle , there's some confusion as to the inner shadow referred to in the above comments. The shadow on the PTR and Swipe Story options should overlay the orange color and not the white. Here's an updated image for reference. Let me know if this doesn't make sense.
![shadow](https://cloud.githubusercontent.com/assets/7061920/2852123/da155716-d12d-11e3-8361-cfe64d39cafb.jpg)

**Comment by ghost** [comment]
@chrismorast 

> Duckmode, do you have everything you need for that? Assets and screenshot is included above in this thread

I've just pushed a build that has a fully functional Duckmode; turns out I had all the necessary assets. Thanks

**Comment by ghost** [comment]
@chrismorast I've just pushed another build that (_I hope_) resolves the issues with the inner shadow on the PTR control. Give it a shot and let me know.

**Comment by chrismorast** [comment]
Nice, thanks @micpringle . One request; can you move the logo/text up by about 60px or 30pt?

Also, I notice that after I focus the search field in DuckMode and hit cancel the logo moves up too far (I guess it shouldn't move at all). Attaching a screenshot.
![photo 2_12](https://cloud.githubusercontent.com/assets/7061920/2853493/ac6f5566-d144-11e3-9ef7-4418ed031757.PNG)

**Comment by ghost** [comment]
@chrismorast I can't replicate the issue of the logo moving. It does move _up_ when the search field has focus, but should move back _down_ when hitting Cancel. Could you try completely removing the app and reinstalling?

![duckmode](https://cloud.githubusercontent.com/assets/20023/2853557/8b16a0f8-d145-11e3-8079-71e6d18efec5.png)

**Comment by chrismorast** [comment]
Thanks @micpringle , that fixed it :+1: 
Can we move the logo and text up on the duck mode screen (60px)?

**Comment by chrismorast** [comment]
@micpringle , in regards to the inner shadows, I like what you had implemented just before this - the fix you came up with. Can we revert that back?

Also, in regards to font size (and color);
Unitl we work out the possible use of Proxima Nova I think what you have is fine with the following exceptions;
1. In the menu, recent search list and recent stories list should be 13pt (the labels at the top of each list are perfect as is)
2. Please update the color of the text in the menu (Not the labels though - ie. RECENT SEARCHES) to #292929. Please update the story titles on the home page to #292929 as well.

One more fix that I can see, can you please add a drop shadow to the bottom story in any list? Attaching how it looks now at the bottom of the home page scroll as well as on the favorites page.
![photo_18](https://cloud.githubusercontent.com/assets/7061920/2855434/6d94a8ec-d15c-11e3-95bd-925a74b36837.PNG)
![photo_19](https://cloud.githubusercontent.com/assets/7061920/2855435/6db71a44-d15c-11e3-8614-84e9e1fb953f.PNG)

**Comment by ghost** [comment]
@chrismorast 

> in regards to the inner shadows, I like what you had implemented just before this - the fix you came up with. Can we revert that back?

Yep. I've reverted back to the previous implementation.

**Comment by ghost** [comment]
@chrismorast 

> Can we move the logo and text up on the duck mode screen (60px)?

This is now in the latest build. Prior to moving it up, it was vertically centred (_even though it didn't appear to be on first sight_). What I've done is move it up 60pts, but thanks to some AutoLayout wizardry, still kept it centre when the keyboard is on screen. It looks much better now. Let me know what you think.

**Comment by ghost** [comment]
@chrismorast 

I've just pushed another build that adds/fixes everything you mention in [this comment](https://github.com/duckduckgo/ios/issues/8#issuecomment-41938599).

**Comment by chrismorast** [comment]
Looks great @micpringle :+1: 

I've notice a few more things that need to be addressed. 
Can you remove the button backgrounds from the keyboard header? Is it possible to use the native version of this?
![button bgs](https://cloud.githubusercontent.com/assets/7061920/2867059/9a005c30-d230-11e3-87ae-ee73eaaea60f.png)

**Comment by chrismorast** [comment]
In the auto suggest screen, can we remove the large search icon in each row and left justify the text suggestions? this would not apply to the !bang auto suggest (which I will be adding a change to next).
![standard autosuggest](https://cloud.githubusercontent.com/assets/7061920/2867086/01e1fc8c-d231-11e3-9757-797cd781c1b1.png)

**Comment by chrismorast** [comment]
Here's what it should look more like. Is it possible to reduce the height of each row as well so we can see more suggestions without scrolling?
![standard autosuggest_update](https://cloud.githubusercontent.com/assets/7061920/2867126/ac6e1cf8-d231-11e3-92fb-6c24a750eac0.png)

**Comment by chrismorast** [comment]
With the !bang autosuggest, can we push the content in each row further to the left?

Here's how it is now:
![bang-autosuggest](https://cloud.githubusercontent.com/assets/7061920/2867615/02a1bc9a-d239-11e3-8727-a445dba7b4e0.png)

Here's how I'd like it to be:
![bang-autosuggest_alignment](https://cloud.githubusercontent.com/assets/7061920/2867620/0bbd1522-d239-11e3-9202-5848a99a4764.png)

**Comment by nilnilnil** [comment]
I'd like to setup a Skype call about NON bang autocomplete.  We need to evaluate middle string ellipsis.

**Comment by chrismorast** [comment]
Next, can we style the saved searches (now favorite searches) to be something like this:
![saved searches](https://cloud.githubusercontent.com/assets/7061920/2867720/535c5644-d23a-11e3-9f54-a94d36497080.jpg)

Also, please note that the tab navigation (on the bottom) should have a drop shadow above it as shown in this comp.

**Comment by chrismorast** [comment]
One last thing that I can see; we need to update the icons in the slide up menu as show here:

![slide-up menu](https://cloud.githubusercontent.com/assets/7061920/2868205/3a7b2bb6-d242-11e3-8b6e-bf0b2c77663b.jpg)

Here are the updated readability assets:
![readability-on 2x](https://cloud.githubusercontent.com/assets/7061920/2868247/fd4e8728-d242-11e3-8643-1bd85eb43df9.png)
![readability-off 2x](https://cloud.githubusercontent.com/assets/7061920/2868248/fd6c3994-d242-11e3-8c64-6497ac10d202.png)
![readability-on](https://cloud.githubusercontent.com/assets/7061920/2868249/fd6e8eba-d242-11e3-8750-c82afca127d7.png)
![readability-off](https://cloud.githubusercontent.com/assets/7061920/2868250/fd70b58c-d242-11e3-846a-18689f9e28dd.png)

**Comment by ghost** [comment]
@chrismorast I've now removed the custom backgrounds from the previous, next and done buttons of the accessory toolbar when filling in forms, such as the Gmail login.

**Comment by ghost** [comment]
@chrismorast The changes you've requested for autocomplete are now in the latest build. Please check them out and let me have any feedback/further changes.

**Comment by nilnilnil** [comment]
My thoughts are in separate tasks.

**Comment by ghost** [comment]
@chrismorast I've updated the sharing icons as per the above. I've also changed the Readability text to read "_Enable Readability_" or "_Disable Readability_" since using on or off implied that was the current state rather than the action. I'd be happy to change it back though if you disagree.

**Comment by ghost** [comment]
@chrismorast Don't forget about the popups that are displayed when favouriting/unfavouriting. See [this comment](https://github.com/duckduckgo/ios/issues/8#issuecomment-41368831) for more.

**Comment by ghost** [comment]
@chrismorast The latest build (_which should be available in a minute or two_) has the updated favourite searches view.

**Comment by chrismorast** [comment]
Hey @micpringle , Here's what I'd like to do for the pop-ups.

![added](https://cloud.githubusercontent.com/assets/7061920/2951902/f2f120a2-da2f-11e3-904d-47dfd927b797.jpg)
![removed](https://cloud.githubusercontent.com/assets/7061920/2951901/f2f0de44-da2f-11e3-8957-936f78454b18.jpg)

**Comment by chrismorast** [comment]
Here are the assets:
![icon_added](https://cloud.githubusercontent.com/assets/7061920/2951915/43d4d22a-da30-11e3-9e10-c7beb0394014.png)
![icon_added 2x](https://cloud.githubusercontent.com/assets/7061920/2951917/43f048ac-da30-11e3-9a1b-1e8304ff3cec.png)
![icon_removed](https://cloud.githubusercontent.com/assets/7061920/2951916/43f0243a-da30-11e3-92ea-1bb4bd910992.png)
![icon_removed 2x](https://cloud.githubusercontent.com/assets/7061920/2951918/43f60c38-da30-11e3-85b0-6571c333a888.png)

**Comment by chrismorast** [comment]
In addition, I'd like to update/try 3 things. 

Can you add a drop shadow on the slide-in menu?
Something like rgba (0,0,0,0.15) with a 40px spread (attaching what that would look like).

Also, when the user swipes to delete can you have the delete button hide the "+" rather than move the "+" over?

Can we try a slight opacity on the header and allow the content to blur as it scrolls behind it?

**Comment by ghost** [comment]
@chrismorast Which header?

**Comment by chrismorast** [comment]
The main header with the menu button, search field, and device status bar.

**Comment by ghost** [comment]
@chrismorast The navigation bar is already translucent and blurs the content below. It's handled automatically by the OS.

**Comment by ghost** [comment]
@chrismorast ~~It appears as though you forgot to attach a screenshot of what the drop shadow on the menu should look like, so for now I've added the same drop shadow that's used on the story cells and navigation bar.~~

I've taken a second stab at the shadow, and now believe we have something much closer to what you had in-mind. Take a look and let me know.

**Comment by ghost** [comment]
@chrismorast 

> Also, when the user swipes to delete can you have the delete button hide the "+" rather than move the "+" over?

This is now fixed in the latest build.

**Comment by ghost** [comment]
@chrismorast The popups have been updated in the latest build. ~~I guess I'm just waiting on feedback/more info regarding the slide-over menu drop shadow now?~~

**Comment by nilnilnil** [comment]
We are all done here. Great work @chrismorast, @infinitekind and @micpringle!

**Comment by chrismorast** [comment]
@micpringle , it looks like the "added" and "removed" pop-ups have been reverted back to the old ones in the newest build. Can you please update? (attaching the desired versions for reference)
![removed](https://cloud.githubusercontent.com/assets/7061920/3057222/b98ffc60-e1d3-11e3-85f2-372fe30e853b.jpg)
![added](https://cloud.githubusercontent.com/assets/7061920/3057223/b9af2e1e-e1d3-11e3-81d1-31212dc4924c.jpg)

One last request on top of that... can you change the color of the placeholder text in the search field (Search DuckDuckGo...) to #808080?

**Comment by ghost** [comment]
Chris,  

Can you try completely deleting the app and reinstalling?  

I've just tried the latest build from Hockey and it has the new pop ups.  

## 

Mic Pringle
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by chrismorast** [comment]
Just uninstalled then installed.When I add a search, the new "added" pop-up shows. However, when I add/remove a story (by swiping it in the story list view) the old pop-ups are used.

Also, can you please confirm the text color change for the placeholder search field text?

![img_3589](https://cloud.githubusercontent.com/assets/7061920/3057450/525d9da6-e1d6-11e3-858d-cf7593a8b245.png)
![img_3588](https://cloud.githubusercontent.com/assets/7061920/3057451/525fc072-e1d6-11e3-8905-424b850e93d3.png)

**Comment by nilnilnil** [comment]
@infinitekind @micpringle reopening based on @chrismorast 's feedback.

**Comment by ghost** [comment]
@chrismorast @nilnilnil 

I've now fixed this issue, and changed the search field placeholder text colour.
